src/Raven.Client/Documents/Session/AsyncSessionDocumentCounters.cs

```diff
@@ -0,0 +1,147 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AsyncSessionDocumentCounters.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Raven.Client.Documents.Session
+{
+    public class AsyncSessionDocumentCounters : SessionCountersBase, ICountersSessionOperationsAsync
+    {
+        public AsyncSessionDocumentCounters(InMemoryDocumentSessionOperations session, string documentId) : base(session, documentId)
+        {
+        }
+
+        public AsyncSessionDocumentCounters(InMemoryDocumentSessionOperations session, object entity) : base(session, entity)
+        {
+        }
+
+        public async Task<Dictionary<string, long>> GetAllAsync()
```

Comment:
CancellationToken here? 
Do we have CT for all our methods?

Reply:
We don't. Wrote a Roslyn analyzer to check that.

Reply:
http://issues.hibernatingrhinos.com/issue/RavenDB-11326